provider/pkg/resources/resources.go

```diff
@@ -119,10 +120,23 @@ type ResourceAutoname struct {
 	ValidationRegex string `json:"validationRegex,omitempty"`
 }
 
+type PollingStrategy string
+
+const (
+	DefaultPoll       = PollingStrategy("")
+	KNativeStatusPoll = PollingStrategy("KNativeStatusPoll")
```

Comment:
We will likely have to support additional custom polling mechanisms here. See https://github.com/GoogleCloudPlatform/declarative-resource-client-library/tree/main/dcl/operations